### PR TITLE
feat(pipelines): add pre-install commands to asset action

### PR DESCRIPTION
**Pull request by hoegertn**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by hoegertn** [comment]
@rix0rrr would love to have this merged as it blocks users running asset uploads in VPCs without open internet access.

File: packages/@aws-cdk/pipelines/lib/pipeline.ts [hunk 1]
```diff
@@ -145,6 +153,27 @@ export interface CdkPipelineProps {
   readonly supportDockerAssets?: boolean;
 }
 
+/**
+ * Optional settings to configure the asset publishing step
+ */
+export interface AssetPublishingOptions {
```
Comment (rix0rrr) [inline packages/@aws-cdk/pipelines/lib/pipeline.ts:159]: Not a big fan of this. I think I'd rather keep the options object flat.

**Review by rix0rrr — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/pipelines/lib/pipeline.ts (hunk 1 above)
Comment (hoegertn) [reply packages/@aws-cdk/pipelines/lib/pipeline.ts:159]: I can add it to the top level object. I was just wondering as we now have at least two settings if we want to group them together as asset stuff.

But if you do not see the need for this I can change it.

Comment (rix0rrr) [reply packages/@aws-cdk/pipelines/lib/pipeline.ts:159]: We have it as a design guideline that we put as much as we can into top-level properties. It's a tough balancing act, but for 2 fields I don't see the need yet.

Comment (hoegertn) [reply packages/@aws-cdk/pipelines/lib/pipeline.ts]: Done

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**